src/sentry/api/endpoints/project_app_store_connect_credentials.py

```diff
@@ -255,18 +251,13 @@ def post(self, request: Request, project: Project) -> Response:
         config["itunesSession"] = session_context.get("itunes_session")
         config["itunesPersonId"] = session_context.get("itunes_person_id")
 
-        # We need to have a serialised datetime, but django-rest-framework de-serialised it
-        # into the class.  So, serialize it back.
-        clone = config.copy()
-        clone["itunesCreated"] = config["itunesCreated"].isoformat()
-        jsonschema.validate(clone, APP_STORE_CONNECT_SCHEMA)
+        validated_config = appconnect.AppStoreConnectConfig.from_json(config)
 
-        # logger.warn("Spawning dsym download task")
         dsym_download.apply_async(
             kwargs={
                 "project_id": project.id,
-                "config_id": config["id"],
-                "config": config,
+                "config_id": validated_config.id,
+                "config": validated_config,
```

Comment:
fwiw this now has to be pickle-safe, so you may never move the class definition around or in-flight tasks break

would do the from_json within the task

Reply:
Speaking of `config`, how big is that object?

Reply:
@untitaker oh, thanks for the hint!  changed that back to be plain json-as-dict

@tonyo my test case is 1924 bytes.  Variation on this will be small, so expect all to be around the 2000 bytes mark.

Reply:
@tonyo this is kind of a work-around around some (now it seems poor) decisions on how to share responsibilities between the UI and the backend.  If you think this is a problem we can completely remove the argument and only send over the two IDs, only cost is that it'll also need some changes to the UI which i was trying to avoid